google-cloud-spanner/src/main/java/com/google/cloud/spanner/SessionPool.java

```diff
@@ -3587,6 +3678,10 @@ public void onSessionReady(SessionImpl sessionImpl) {
     public void onSessionCreateFailure(Throwable t, int createFailureForSessionCount) {
       synchronized (lock) {
         multiplexedSessionBeingCreated = false;
+        if (isDatabaseOrInstanceNotFound(newSpannerException(t))) {
```

Comment:
nit: use `asSpannerException(t)` instead of `newSpannerException(t)`. That will keep the original exception if it was already a `SpannerException`.

Reply:
Done